### PR TITLE
Allow custom plugins when running globally

### PR DESCRIPTION
**Pull request by pvdlg**
Previously the plugins were imported relatively to `process.cwd()`.
If a custom plugin is defined in `package.json` via a globally installed npm package it would fails even when `semantic-release` is installed and ran globally.

This PR allow to load plugin:
- As an npm packaged depended on in `package.json` when run locally or globally
- A `.js` file, referenced relatively to the root of the repo, when run locally or globally
- A globally installed npm package when run globally

`semantic-release` can now be installed and ran globally, along with globally installed custom plugins without having to be included as a dependency in the `package.json`. It might be convenient for a future Github app :)

### CODE REVIEW

**Comment by gr2m** [comment]
Is this fixing an open issue that we have? I don’t quite understand the use case yet :) Can you explain a bit further?

**Comment by pvdlg** [comment]
No we don't have an open issue regarding that, but it was mentioned in several conversation that it would be nice to be able to run `semantic-release` globally without having to depend on it in the `package.json`

What this change allow to do is to have in package.json:
```json
"release": {
  "analyzeCommits": "my-custom-analyzer"
}
```
and no dependencies to `semantic-release` or `my-custom-analyzer` in `devDependencies` and be able to run:
```bash
npm install -g semantic-release my-custom-analyzer
semantic-release pre && npm publish && semantic-release post
```

One possible benefit would be to not have the problem with `yarn` that refuses to install `semantic-release` as a `devDependency` on version of Node < 8 and would avoid to have to do the trick with `"engines": {"node": "4"}`.

Another possible advantage is the situation in which a release is triggered externally (for example when a Github app would detect via commit status that all CI pass successfully). You can trigger a job that install `semantic-release` globally and run. So for the user no need to include and manage `semantic-release` and custom plugins as a `devDependency`.

In conclusion, this is useful only when we run `semantic-release` globally and it's something that we don't officially recommend in the doc for now. But it might be useful in the future.

**Comment by gr2m** [comment]
I fear that we introduce a functionality which we don’t yet have use cases for and that we will have to maintain for forever. Maybe let’s leave that open until we hit a blocker that this is the right solution for?

**Comment by pvdlg** [comment]
Ok, works for me

**Comment by pvdlg** [comment]
Irrelevant due to #524
